docs/architecture/adr-060-baseapp++.md

```diff
@@ -0,0 +1,242 @@
+ignore _all_ transactions sent to it in the request and instead reap up to
+`PrepareProposalRequest.max_tx_bytes` from it's own mempool. There is no need to
+execute the transactions for validity as they have already passed CheckTx.
+
+### `ProcessProposal`
+
+
+<!-- 
+## Consequences
+
+> This section describes the resulting context, after applying the decision. All consequences should be listed here, not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences, but all of them affect the team and project in the future.
+
+### Backwards Compatibility
+
+> All ADRs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The ADR must explain how the author proposes to deal with these incompatibilities. ADR submissions without a sufficient backwards compatibility treatise may be rejected outright.
+
+### Positive
+
+{positive consequences}
+
+### Negative
+
+{negative consequences}
+
+### Neutral
+
+{neutral consequences}
+
+## Further Discussions
+
+While an ADR is in the DRAFT or PROPOSED stage, this section should contain a summary of issues to be solved in future iterations (usually referencing comments from a pull-request discussion).
+Later, this section can optionally list ideas or improvements the author or reviewers found during the analysis of this ADR. -->
+
+## References
+
+* https://github.com/tendermint/tendermint/blob/master/spec/abci%2B%2B/README.md
+* [1] https://github.com/tendermint/tendermint/issues/7750#issuecomment-1076806155
+* [2] https://github.com/tendermint/tendermint/issues/7750#issuecomment-1075717151
+
+
+<!-- ----------------------------------------------------------------------- -->
+<!-- ----------------------------------------------------------------------- -->
+<!-- ----------------------------------------------------------------------- -->
+
+<!-- 
+### Prepare Proposal
+
+There are two designs being considered for `Prepare Proposal`:
+
+#### App-defined Mempool
+
+An app defined mempool is a mempool that lives within the application. Currently Tendermint houses all the ordering and block creation logic. While this abstractions simplifies applications it prohibits many usecases from being achieved. 
+
+The appside mempool is populated via `Checktx`. Tendermint will continue calling `Checktx` for the application to verify the validity of a transaction. When the application receives the transaction, it will add it to its own mempool. While the app is continuing the app will have the option to modify or order transactions within any preference it would like. The ordering of transactions is optional for full nodes. When a block is being proposed, Tendermint will send an array of bytes to the application and ask if the bytes are to be included in the block or replaced with transactions the app would like. At this time the application can replace, modify or delete transcations from the array of bytes. 
```

Comment:
> While the app is continuing the app will have the option to modify or order transactions within any preference it would like

Consider being specific that modifications happen in `PrepareProposal` since `CheckTx` is also referenced above and might confuse the reader